gateway-provider-identity-assertion-common/src/main/java/org/apache/knox/gateway/identityasserter/common/filter/CommonIdentityAssertionFilter.java

```diff
@@ -17,30 +17,45 @@
  */
 package org.apache.knox.gateway.identityasserter.common.filter;
 
+import java.io.IOException;
+import java.security.AccessController;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 import javax.security.auth.Subject;
 import javax.servlet.FilterChain;
 import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletRequestWrapper;
 
 import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.knox.gateway.IdentityAsserterMessages;
 import org.apache.knox.gateway.i18n.messages.MessagesFactory;
+import org.apache.knox.gateway.plang.Ast;
+import org.apache.knox.gateway.plang.Parser;
+import org.apache.knox.gateway.plang.SyntaxException;
+import org.apache.knox.gateway.security.GroupPrincipal;
 import org.apache.knox.gateway.security.principal.PrincipalMappingException;
 import org.apache.knox.gateway.security.principal.SimplePrincipalMapper;
 
-import java.io.IOException;
-import java.security.AccessController;
-
 public class CommonIdentityAssertionFilter extends AbstractIdentityAssertionFilter {
+  public static final String VIRTUAL_GROUP_MAPPING_PREFIX = "virtual.group.mapping.";
```

Comment:
I thought we agreed to remove the "virtual" language from the configuration. Did we revert that agreement?
I think it's fine in the code, but from a configuration-perspective, is there anything preventing mapping a user to a "real" group with this mechanism?

Reply:
I removed the virtual prefix, now it's just "group.mapping"

The provider supports the following:

* group.principal.mapping -> kminder=group1;lmccay=mrgroup,mrducks
* principal.mapping -> lmccay,kminder=hdfs;newuser=mapred
* group.mapping -> (username 'admin')

The 3rd one is the new one.